tests/Unit/Infrastructure/Persistence/IlluminateSpeakerRepositoryTest.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace OpenCFP\Test\Unit\Infrastructure\Persistence;
+
+use Illuminate\Database\Eloquent;
+use OpenCFP\Domain\EntityNotFoundException;
+use OpenCFP\Domain\Model;
+use OpenCFP\Domain\Speaker;
+use OpenCFP\Infrastructure\Persistence\IlluminateSpeakerRepository;
+use OpenCFP\Test\Helper\Faker\GeneratorTrait;
+use PHPUnit\Framework;
+
+/**
+ * @covers \OpenCFP\Infrastructure\Persistence\IlluminateSpeakerRepository
+ */
+final class IlluminateSpeakerRepositoryTest extends Framework\TestCase
+{
+    use GeneratorTrait;
+
+    public function testImplementsSpeakerRepository()
+    {
+        $reflection = new \ReflectionClass(IlluminateSpeakerRepository::class);
+
+        $this->assertTrue($reflection->implementsInterface(Speaker\SpeakerRepository::class));
+    }
+
+    public function testFindByIdThrowsEntityNotFoundExceptionIfUserNotFound()
+    {
+        $id = $this->getFaker()->numberBetween(1);
+
+        $userModel = $this->createUserMock([
+            'findOrFail',
+        ]);
+
+        $userModel
+            ->expects($this->once())
+            ->method('findOrFail')
+            ->with($this->identicalTo($id))
+            ->willThrowException(new Eloquent\ModelNotFoundException());
+
+        $repository = new IlluminateSpeakerRepository($userModel);
+
+        $this->expectException(EntityNotFoundException::class);
+
+        $repository->findById($id);
+    }
+
+    public function testFindByIdReturnsUser()
+    {
+        $id = $this->getFaker()->numberBetween(1);
+
+        $user = $this->createUserMock();
+
+        $userModel = $this->createUserMock([
+            'findOrFail',
+        ]);
+
+        $userModel
+            ->expects($this->once())
+            ->method('findOrFail')
+            ->with($this->identicalTo($id))
+            ->willReturn($user);
+
+        $repository = new IlluminateSpeakerRepository($userModel);
+
+        $this->assertSame($user, $repository->findById($id));
+    }
+
+    public function testPersistSavesUser()
+    {
+        $user = $this->createUserMock([
+            'save',
+        ]);
+
+        $user
+            ->expects($this->once())
+            ->method('save');
+
+        $repository = new IlluminateSpeakerRepository($this->createUserMock());
+
+        $repository->persist($user);
```

Comment:
I am not sure what [`IlluminateSpeakerRepository::persist()`](https://github.com/opencfp/opencfp/blob/70b5aff3fb766b38642a6f2ae3f350871d740fae/classes/Infrastructure/Persistence/IlluminateSpeakerRepository.php#L49-L52) is supposed to return, as the method `Model\User::save()` appears to be a magic method.

Reply:
If I recall the pattern correctly, the persist() method should return a boolean value based on whether or not it was able to create a record in the database for $user

Reply:
yes the [save()](https://laravel.com/api/5.5/Illuminate/Database/Eloquent/Model.html#method_save function returns a boolean, whether it was successful or not.

Reply:
Thank you, @BackEndTea! 

I don't think we really need to return a `bool` here, what do you think?